ecs.tf

```diff
@@ -59,43 +51,81 @@ resource "aws_appautoscaling_target" "app_scale_target" {
   min_capacity       = var.ecs_autoscale_min_instances
 }
 
+module "app_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = var.container_name
+  container_image              = var.docker_image
+  container_cpu                = var.container_cpu
+  container_memory             = var.container_memory
+  essential                    = true
+  readonly_root_filesystem     = false
+  environment                  = var.environment_vars
+  port_mappings                = [
+    {
+      containerPort            = var.container_port
+      hostPort                 = var.container_port
+      protocol                 = "tcp"
+    }
+  ]
+  log_configuration            = {
+    logDriver = "awslogs"
+    options   = {
+      "awslogs-group" = "/fargate/service/${var.app}-${var.environment}"
+      "awslogs-region" = "eu-west-1"
+      "awslogs-stream-prefix" = "ecs"
+    }
+    secretOptions = null
+  }
+}
+
+module "datadog_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = "datadog-agent"
+  container_image              = "datadog/agent:latest"
+  container_cpu                = "10"
+  container_memory             = "128"
+  essential                    = false
+  readonly_root_filesystem     = false
+  environment                  = [
+   {
+      name = "DD_API_KEY",
+      value = var.datadog_api_key
```

Comment:
I'm afraid by doing this you will make this secret appear as a plain text in the plan output, so this isn't a good idea to inject it that way.